lib/timespan.py

```diff
@@ -0,0 +1,89 @@
+
+        ts_id = '{year1} - {year2}'.format(year1=year1, year2=year2)
+        ts_name = '{year1} - {year2}'.format(year1=year1, year2=year2)
+        ts_description = 'From {year1} to {year2}'.format(year1=year1, year2=year2)
+        ts_timespan = '[{year1}-01-01, {year2}-12-31]'.format(year1=year1, year2=year2)
+    # Year and month, for example '2017-03'
+    elif re.match('\d{4}-\d{2}$', timespan_str):
+        year, month = list(map(int, timespan_str.split('-')))
+        if month > 12:
+            raise ValueError("Invalid timespan '{timespan}' (invalid month)".format(timespan=timespan_id))
+
+        last_day_of_month = calendar.monthrange(year, month)[1]
+        month_name = calendar.month_name[month]
+
+        ts_id = '{year}-{month:02d}'.format(year=year, month=month)
+        ts_name = '{year}-{month:02d}'.format(year=year, month=month)
+        ts_description = '{month_name} {year}'.format(month_name=month_name, year=year)
+        ts_timespan = '[{year}-{month:02d}-01, {year}-{month:02d}-{last_day}]'.format(year=year, month=month,
+                                                                                      last_day=last_day_of_month)
+    # Year and quarter, for example '2017Q3'
+    elif re.match('\d{4}Q\d{1}$', timespan_str):
+        year, quarter = list(map(int, timespan_str.split('Q')))
+        if quarter > 4:
+            raise ValueError("Invalid timespan '{timespan}' (invalid quarter)".format(timespan=timespan_id))
+
+        last_month = quarter * 3
+        initial_mont = last_month - 2
+        last_day = calendar.monthrange(year, last_month)[1]
+
+        ts_id = '{year}Q{quarter}'.format(year=year, quarter=quarter)
+        ts_name = '{year}Q{quarter}'.format(year=year, quarter=quarter)
+        ts_description = '{year}, {quarter} quarter'.format(year=year, quarter=QUARTERS[quarter])
+        ts_timespan = '[{year}-{ini_month:02d}-01, {year}-{end_month:02d}-{last_day}]'.format(year=year,
+                                                                                              ini_month=initial_mont,
+                                                                                              end_month=last_month,
+                                                                                              last_day=last_day)
+    # Day, month, year, for example '20171231'
+    elif re.match('\d{8}$', timespan_str):
+        year = timespan_str[:4]
+        month = timespan_str[4:6]
+        day = timespan_str[-2:]
+
+        if int(month) > 12:
+            raise ValueError("Invalid timespan '{timespan}' (invalid month)".format(timespan=timespan_id))
+
+        last_day_of_month = calendar.monthrange(int(year), int(month))[1]
+        if int(day) > last_day_of_month:
+            raise ValueError("Invalid timespan '{timespan}' (invalid day)".format(timespan=timespan_id))
+
+        ts_id = '{year}{month}{day}'.format(year=year, month=month, day=day)
```

Comment:
Using the `datetime` object returned by the `parse` object, you could use 'd.strftime('%Y%m%d')` to get the final attributes using the desired format